### DISCUSSION TITLE
How to splice screenshots

### DISCUSSION THREAD

**Discussion by z2labplus** [OpenStitching/stitching#110] [Q&A]
![1689444989330](https://github.com/OpenStitching/stitching/assets/8223359/57fc3ba4-5bcf-461a-ad69-f6baf4ebd1e4)
The error
In [16]: imgs = ['1.jpg','2.jpg','3.jpg']

In [17]: stitcher = stitching.Stitcher()

In [18]: panorama = stitcher.stitch(["img1.jpg", "img2.jpg", "img3.jpg"])
---------------------------------------------------------------------------
StitchingError                            Traceback (most recent call last)
<ipython-input-18-b6dbb63fadb0> in <module>
----> 1 panorama = stitcher.stitch(["img1.jpg", "img2.jpg", "img3.jpg"])

/usr/local/lib/python3.10/dist-packages/stitching/stitcher.py in stitch(self, img_names)
     92         imgs, features, matches = self.subset(imgs, features, matches)
     93         cameras = self.estimate_camera_parameters(features, matches)
---> 94         cameras = self.refine_camera_parameters(features, matches, cameras)
     95         cameras = self.perform_wave_correction(cameras)
     96         self.estimate_scale(cameras)

/usr/local/lib/python3.10/dist-packages/stitching/stitcher.py in refine_camera_parameters(self, features, matches, cameras)
    145 
    146     def refine_camera_parameters(self, features, matches, cameras):
--> 147         return self.camera_adjuster.adjust(features, matches, cameras)
    148 
    149     def perform_wave_correction(self, cameras):

/usr/local/lib/python3.10/dist-packages/stitching/camera_adjuster.py in adjust(self, features, pairwise_matches, estimated_cameras)
     47         b, cameras = self.adjuster.apply(features, pairwise_matches, estimated_cameras)
     48         if not b:
---> 49             raise StitchingError("Camera parameters adjusting failed.")
     50 
     51         return cameras

StitchingError: Camera parameters adjusting failed.

The 1.jpg 2.jpg 3.jpg is the three screenshot
I want to stitch together these three cell phone screenshots into one picture
But it the stitching return me the error above
Please help me

**Comment by lukasalexanderweber** [comment]
You need to use the `AffineStitcher` for Screenshots

**Reply by z2labplus** [reply]
how to use it,any demo

**Reply by lukasalexanderweber** [reply]
https://github.com/OpenStitching/stitching_tutorial/blob/master/docs/Stitching%20Tutorial.md#affine-stitcher

**Reply by z2labplus** [reply]
I tried AffineStitcher , but it didn't work

**Reply by lukasalexanderweber** [reply]
Can you share the Images?

**Reply by z2labplus** [reply]
the 1 2 3 jpg, and Vertical splicing

**Reply by lukasalexanderweber** [reply]
This wont work. Use a Photo Editor instead

**Reply by z2labplus** [reply]
what Photo Editor  ?
